### PR TITLE
Added zoom plugin

### PR DESCRIPTION
**Pull request by CBarrett2**
zoomPlugin plugin adds zoom and pan functionality - still needs to fix scaleExtent, so you don't zoom off the chart.

### CODE REVIEW

File: src/assets/js/chartFramework.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+(function(fcsc, fc, d3) {
+    'use strict';
+    fcsc.chartFramework = function() {
+        var width = 500;
+        var height = 300;
+        var timeSeries = fc.chart.linearTimeSeries();
```
Comment (jleft) [inline src/assets/js/chartFramework.js]: `linearTimeSeries` in more than just a series, so maybe rename `timeSeries` to `chart`?

File: src/assets/js/chartFramework.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        var multi = fc.series.multi();
+        var plugins = [];
+        var dateScale = fc.scale.dateTime()
```
Comment (jleft) [inline src/assets/js/chartFramework.js]: Do we need this? `linearTimeSeries` has an `xScale` property; are we effectively duplicating some of its behavior?

For example, if we need the x-scale, could we just do?

```js
timeSeries.xScale();
```

File: src/assets/js/chartFramework.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            .range([0, width]);
+        var scaleExtent = [];
```
Comment (jleft) [inline src/assets/js/chartFramework.js]: Is `scaleExtent` used?

File: src/assets/js/chartFramework.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+        function chart(selection) {
+            selection.each(function(d, i) {
+                scaleExtent = fc.util.extent(d, ['date']);
+
+                // Y-axis automatically scales
+                timeSeries.xDomain(dateScale.domain())
+                    .yDomain(fc.util.extent(d, ['open', 'close']));
```
Comment (jleft) [inline src/assets/js/chartFramework.js]: This assumes that the data will have open/close properties, and that the series will be created with y-values of either the open or close -  which may not always be the case.

For example, if the user specifies a series to use the data's `high` property, some of the data may be clipped:

```js
var line = fc.series.line()
    .yValue(function(d) { return d.high; });
```

File: src/assets/js/chartFramework.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        };
+
+        chart.dateScale = function(value) {
+            if (!arguments.length) { return dateScale; }
+            dateScale = value;
+            return chart;
+        };
+
+        // Perhaps this should be done using d3.rebind
+        chart.initDomain = function(value) {
```
Comment (jleft) [inline src/assets/js/chartFramework.js]: Do we need a setter for the init domain, or could this be set outside of chartFramework and passed in as a property of the scale?

File: src/assets/js/chartFramework.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            if (!arguments.length) { return dateScale.domain(); }
+            dateScale.domain(value);
+            return chart;
+        };
+
+        chart.width = function(value) {
+            if (!arguments.length) { return width; }
+            width = value;
+            dateScale.range([0, width]);
```
Comment (jleft) [inline src/assets/js/chartFramework.js]: Can we set this in `function chart { ...`?

File: src/assets/js/chartFramework.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+        chart.plugins = function(value) {
+            if (!arguments.length) { return plugins; }
+            plugins = value;
+            return chart;
+        };
+
+        d3.rebind(chart, multi, 'series');
+
+        chart.getVisibleData = function(data, dateExtent) { // currently unused
```
Comment (jleft) [inline src/assets/js/chartFramework.js]: If this is unused, shall we remove it and add it back in later?

File: src/assets/js/zoomPlugin.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+(function(fcsc, fc, d3) {
+    'use strict';
+    fcsc.zoomPlugin = function() {
```
Comment (jleft) [inline src/assets/js/zoomPlugin.js]: `d3.behavior.zoom()` is effectively a 'plugin', do we need to wrap it?

**Comment by jleft** [comment]
Looks like a really good start for now, thanks @CBarrett2 :)

File: src/assets/js/chartFramework.js (hunk 1 above)
Comment (CBarrett2) [inline src/assets/js/chartFramework.js]: We already have a function named chart though - perhaps something else? seriesChart?

File: src/assets/js/chartFramework.js (hunk 6 above)
Comment (CBarrett2) [inline src/assets/js/chartFramework.js]: How do you mean? Like, as a default value, or that it reads the size of the parent element?

File: src/assets/js/zoomPlugin.js (hunk 8 above)
Comment (CBarrett2) [inline src/assets/js/zoomPlugin.js]: I've added more functionality in the callback now making me think it's neater - check the new pull request :)

**Pull request closed without merging**